kevm_pyk/solc_to_k.py

```diff
@@ -192,16 +193,24 @@ def recur_mapping(syntax, lhs, rhs, var_idx, key_type_name, value_type_name):
     return recur_struct([], f'{contract_name}', _intToken('0'), 0, storage, gen_dot=False)
 
 
-def generate_function_sentences(contract_name, contract_sort, abi):
+def generate_function_sentences(contract_name, contract_sort, abi, hashes):
     function_sort = KSort(f'{contract_name}Function')
+    function_call_data_production = KProduction([KNonTerminal(contract_sort), KTerminal('.'), KNonTerminal(function_sort)], KSort('ByteArray'), att=KAtt({'klabel': f'function_{contract_name}', 'function': ''}))
     function_sentence_pairs = _extract_function_sentences(contract_name, function_sort, abi)
 
     if not function_sentence_pairs:
         return []
 
+    abi_function_selector_production = KProduction([KTerminal('selector'), KTerminal('('), KNonTerminal(KSort('String')), KTerminal(')')], KSort('Int'), att=KAtt({'klabel': 'abi_selector', 'alias': ''}))
+    abi_function_selector_rules = []
+    for h in hashes:
+        f_name   = h.split('(')[0]
+        hash_int = int(hashes[h], 16)
+        abi_function_selector_rewrite = KRewrite(KToken(f'selector("{f_name}")', function_sort['name']), KToken(str(hash_int), 'Int'))
```

Comment:
:question: Should it not be `Int` instead?

```suggestion
        abi_function_selector_rewrite = KRewrite(KToken(f'selector("{f_name}")', KSort('Int')), KToken(str(hash_int), 'Int'))
```

Reply:
Fixed.